patches/api/0441-Add-predicate-for-blocks-when-raytracing.patch

```diff
@@ -0,0 +1,119 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TonytheMacaroni <tonythemacaroni123@gmail.com>
+Date: Wed, 6 Sep 2023 19:24:53 -0400
+Subject: [PATCH] Add predicate for blocks when raytracing
+
+
+diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
+index da524a71af74b02515b037f7fe09ba6988e2c8bf..0b4001be5b0f2a5d2b57c1afe77104ad724c0ed6 100644
+--- a/src/main/java/org/bukkit/World.java
++++ b/src/main/java/org/bukkit/World.java
+@@ -1713,6 +1713,28 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
+     @Nullable
+     public RayTraceResult rayTraceEntities(@NotNull Location start, @NotNull Vector direction, double maxDistance, double raySize, @Nullable Predicate<Entity> filter);
+ 
++    // Paper Start
++    /**
++     * Performs a ray trace that checks for entity collisions.
++     * <p>
++     * This may not consider entities in currently unloaded chunks. Some
++     * implementations may impose artificial restrictions on the maximum
++     * distance.
++     *
++     * @param start the start position
++     * @param direction the ray direction
++     * @param maxDistance the maximum distance
++     * @param raySize entity bounding boxes will be uniformly expanded (or
++     *     shrinked) by this value before doing collision checks
++     * @param filter only entities that fulfill this predicate are considered,
++     *     or <code>null</code> to consider all entities
++     * @return the closest ray trace hit result, or <code>null</code> if there
++     *     is no hit
++     */
++    @Nullable
++    public RayTraceResult rayTraceEntities(@NotNull io.papermc.paper.math.Position start, @NotNull Vector direction, double maxDistance, double raySize, @Nullable Predicate<Entity> filter);
++    // Paper End
```

Comment:
Don't need to add this. We will change Location to Position at some point in the future.

Reply:
Required because `CraftWorld#rayTrace` calls `CraftWorld#rayTraceEntities` with its start position. Could construct a location from the position, but that seems a bit jank.